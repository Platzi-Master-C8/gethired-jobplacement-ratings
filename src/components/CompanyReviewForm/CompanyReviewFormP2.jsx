import * as React from 'react';
import PropTypes from 'prop-types';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import {
    Accordion,
    AccordionSummary,
    AccordionDetails,
    FormControl,
    OutlinedInput,
    Grid,
    Typography,
    FormControlLabel,
    RadioGroup,
    Radio,
    Select,
    MenuItem,
    Checkbox,
} from '@mui/material';

import { DateInput } from '../../controls/DateInput';

const CompanyReviewFormP2 = ({ review, handleInput }) => {
    const handleCheck = (val) => {
        const e = { target: { value: val.target.checked ? 1 : 0, name: 'isStillWorkingHere' } };
        handleInput(e);
    };

    return (
        <Accordion sx={{ boxShadow: 'none', marginTop: '16px' }}>
            <AccordionSummary
                expandIcon={<ExpandMoreIcon />}
                aria-controls="additional-quiestion-content"
                id="additional-quiestion-header"
            >
                <Typography variant="h1" align="center" sx={{ fontSize: 28 }}>
                    Additional Questions (Optional)
                </Typography>
            </AccordionSummary>
            <AccordionDetails>
                <Grid item md={12} marginBottom={3}>
                    <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                        Job Location
                    </Typography>
                    <FormControl sx={{ width: '100%' }} required>
                        <OutlinedInput
                            id="jobLocation"
                            name="jobLocation"
                            placeholder="Job Location"
                            value={review.jobLocation}
                            onChange={handleInput}
                        />
                    </FormControl>
                </Grid>
                <Grid container marginBottom={2} spacing={2}>
                    <Grid item md={6}>
                        <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                            Start Date
                        </Typography>
                        <FormControl sx={{ width: '100%' }} required>
                            <DateInput
                                id="startDate"
                                name="startDate"
                                value={review.startDate}
                                onChange={handleInput}
                            />
                        </FormControl>
                    </Grid>
                    <Grid item md={6}>
                        <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                            End Date
                        </Typography>
                        <FormControl sx={{ width: '100%' }} required>
                            <DateInput
                                id="endDate"
                                name="endDate"
                                value={review.endDate}
                                disabled={!!review.isStillWorkingHere}
                                onChange={handleInput}
                            />
                        </FormControl>
                    </Grid>
                </Grid>
                <Grid container marginBottom={3}>
                    <FormControlLabel
                        control={<Checkbox onChange={handleCheck} checked={!!review.isStillWorkingHere} />}
                        label="Is still working here?"
                    />
                </Grid>
                <Grid container spacing={2}>
                    <Grid item md={6}>
                        <Grid item md={12} marginBottom={3}>
                            <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                                Whats the salary?
                            </Typography>
                            <FormControl sx={{ width: '40%' }} required>
                                <OutlinedInput
                                    id="salary"
                                    name="salary"
                                    placeholder="$"
                                    value={review.salary}
                                    onChange={handleInput}
                                />
                            </FormControl>
                            <FormControl sx={{ width: '30%' }} required>
                                <OutlinedInput
                                    id="currencyType"
                                    name="currencyType"
                                    placeholder="MXN"
                                    value={review.currencyType}
                                    onChange={handleInput}
                                />
                            </FormControl>
                            <FormControl sx={{ width: '30%' }} required>
                                <Select
                                    labelId="salaryFrequency-label"
                                    id="salaryFrequency"
                                    name="salaryFrequency"
                                    value={review.salaryFrequency}
                                    label="Per"
                                    onChange={handleInput}
                                >
                                    <MenuItem value={0}>Year</MenuItem>
                                    <MenuItem value={1}>Day</MenuItem>
                                    <MenuItem value={2}>Hour</MenuItem>
                                </Select>
                            </FormControl>
                        </Grid>
                        <Grid item md={12}>
                            <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                                Would you recommend working here to a friend?
                            </Typography>
                            <FormControl component="fieldset" required>
                                <RadioGroup
                                    id="recommendedAFriend"
                                    name="recommendedAFriend"
                                    row
                                    value={review.recommendedAFriend}
                                    onChange={handleInput}
                                >
                                    <FormControlLabel value={1} control={<Radio />} label="Yes" />
                                    <FormControlLabel value={0} control={<Radio />} label="No" />
                                </RadioGroup>
                            </FormControl>
                        </Grid>
                    </Grid>
                    <Grid item md={6}>
                        <Grid item md={12} marginBottom={3}>
                            <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                                Does the job allows you to work from home?
                            </Typography>
                            <FormControl component="fieldset" required>
                                <RadioGroup
                                    id="allowsRemoteWork"
                                    name="allowsRemoteWork"
                                    row
                                    value={review.allowsRemoteWork}
                                    onChange={handleInput}
                                >
                                    <FormControlLabel value={1} control={<Radio />} label="Yes" />
                                    <FormControlLabel value={0} control={<Radio />} label="No" />
                                </RadioGroup>
                            </FormControl>
                        </Grid>
                        <Grid item md={12}>
                            <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                                Is the company legally regulated?
                            </Typography>
                            <FormControl component="fieldset" required>
                                <RadioGroup
                                    id="isLegallyCompany"
                                    name="isLegallyCompany"
                                    row
                                    value={review.isLegallyCompany}
                                    onChange={handleInput}
                                >
                                    <FormControlLabel value={1} control={<Radio />} label="Yes" />
                                    <FormControlLabel value={0} control={<Radio />} label="No" />
                                </RadioGroup>
                            </FormControl>
                        </Grid>
                    </Grid>
                </Grid>
            </AccordionDetails>
        </Accordion>
    );
};

CompanyReviewFormP2.propTypes = {
    review: PropTypes.shape({
        companyId: PropTypes.number.isRequired,
        userId: PropTypes.number.isRequired,
        jobTitle: PropTypes.string.isRequired,
        contentType: PropTypes.string.isRequired,
        careerDevelopmentRating: PropTypes.string.isRequired,
        diversityEqualOpportunityRating: PropTypes.string.isRequired,
        workingEnvironmentRating: PropTypes.string.isRequired,
        salaryRating: PropTypes.string.isRequired,
        jobLocation: PropTypes.string.isRequired,
        startDate: PropTypes.string.isRequired,
        endDate: PropTypes.string.isRequired,
        isStillWorkingHere: PropTypes.oneOf([0, 1]).isRequired,
        salary: PropTypes.number.isRequired,
        currencyType: PropTypes.string.isRequired,
        salaryFrequency: PropTypes.string.isRequired,
        recommendedAFriend: PropTypes.oneOf([0, 1]).isRequired,
        allowsRemoteWork: PropTypes.oneOf([0, 1]).isRequired,
        isLegallyCompany: PropTypes.oneOf([0, 1]).isRequired,
        utilityCounter: PropTypes.number.isRequired,
        nonUtilityCounter: PropTypes.number.isRequired,
        createdAt: PropTypes.string.isRequired,
    }).isRequired,
    handleInput: PropTypes.func.isRequired,
};

export default CompanyReviewFormP2;
