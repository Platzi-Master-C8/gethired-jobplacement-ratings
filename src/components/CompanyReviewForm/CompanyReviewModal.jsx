import React from 'react';
import PropTypes from 'prop-types';
import {
    Alert,
    Box,
    Button,
    Checkbox,
    FormControl,
    FormControlLabel,
    Grid,
    MenuItem,
    Modal,
    RadioGroup,
    Radio,
    Select,
    OutlinedInput,
    Typography,
} from '@mui/material';
import { DateInput } from '../../controls/DateInput';

const boxStyles = {
    position: 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    width: '65%',
    height: '75%',
    bgcolor: 'white',
    boxShadow: 24,
    p: 4,
    overflow: 'auto',
};

const CompanyReviewForm = ({ open, handleClose, handleValidate, modalError, review, handleInput, handleCheck }) => (
    <Modal
        open={open}
        onClose={handleClose}
        aria-labelledby="modal-modal-title"
        aria-describedby="modal-modal-description"
    >
        <form onSubmit={handleValidate}>
            <Box sx={boxStyles}>
                <Typography variant="h1" align="center" sx={{ fontSize: 28 }}>
                    Review the Company
                </Typography>
                <Grid container marginTop={5}>
                    <Grid item md={12} marginBottom="20px">
                        <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                            Job Title
                        </Typography>
                        <FormControl sx={{ width: '100%' }} required>
                            <OutlinedInput
                                id="job_title"
                                name="job_title"
                                placeholder="Job Title"
                                value={review.job_title}
                                onChange={handleInput}
                            />
                        </FormControl>
                    </Grid>
                    <Grid item md={12} marginBottom="20px">
                        <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                            Your Email
                        </Typography>
                        <FormControl sx={{ width: '100%' }} required>
                            <OutlinedInput
                                id="applicant_email"
                                name="applicant_email"
                                placeholder="Your Email"
                                value={review.applicant_email}
                                onChange={handleInput}
                            />
                        </FormControl>
                    </Grid>
                    <Grid item md={12} marginBottom="20px">
                        <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                            Content
                        </Typography>
                        <FormControl sx={{ width: '100%' }} required>
                            <OutlinedInput
                                id="content_type"
                                name="content_type"
                                placeholder="Content"
                                value={review.content_type}
                                onChange={handleInput}
                            />
                        </FormControl>
                    </Grid>
                </Grid>
                <Typography variant="h1" align="center" sx={{ fontSize: 28 }}>
                    How would you rate the following areas ?
                </Typography>
                <Grid marginTop={3} marginBottom={3}>
                    <Grid container sx={{ alignItems: 'center' }}>
                        <Grid item md={6}>
                            <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                                Carrer development
                            </Typography>
                        </Grid>
                        <Grid item md={6}>
                            <FormControl component="fieldset">
                                <RadioGroup
                                    id="career_development_rating"
                                    data-testid="career_development_rating"
                                    name="career_development_rating"
                                    row
                                    value={review.career_development_rating}
                                    onChange={handleInput}
                                >
                                    <FormControlLabel value="Good" control={<Radio />} label="Good" />
                                    <FormControlLabel value="Regular" control={<Radio />} label="Regular" />
                                    <FormControlLabel value="Bad" control={<Radio />} label="Bad" />
                                </RadioGroup>
                            </FormControl>
                        </Grid>
                    </Grid>
                    <Grid container sx={{ alignItems: 'center' }}>
                        <Grid item md={6}>
                            <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                                Diversity and Equal Opportunity
                            </Typography>
                        </Grid>
                        <Grid item md={6}>
                            <FormControl component="fieldset">
                                <RadioGroup
                                    id="diversity_equal_opportunity_rating"
                                    data-testid="diversity_equal_opportunity_rating"
                                    name="diversity_equal_opportunity_rating"
                                    row
                                    value={review.diversity_equal_opportunity_rating}
                                    onChange={handleInput}
                                >
                                    <FormControlLabel value="Good" control={<Radio />} label="Good" />
                                    <FormControlLabel value="Regular" control={<Radio />} label="Regular" />
                                    <FormControlLabel value="Bad" control={<Radio />} label="Bad" />
                                </RadioGroup>
                            </FormControl>
                        </Grid>
                    </Grid>
                    <Grid container sx={{ alignItems: 'center' }}>
                        <Grid item md={6}>
                            <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                                Working Environment
                            </Typography>
                        </Grid>
                        <Grid item md={6}>
                            <FormControl component="fieldset">
                                <RadioGroup
                                    id="working_environment_rating"
                                    data-testid="working_environment_rating"
                                    name="working_environment_rating"
                                    row
                                    value={review.working_environment_rating}
                                    onChange={handleInput}
                                >
                                    <FormControlLabel value="Good" control={<Radio />} label="Good" />
                                    <FormControlLabel value="Regular" control={<Radio />} label="Regular" />
                                    <FormControlLabel value="Bad" control={<Radio />} label="Bad" />
                                </RadioGroup>
                            </FormControl>
                        </Grid>
                    </Grid>
                    <Grid container sx={{ alignItems: 'center' }}>
                        <Grid item md={6}>
                            <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                                Salary
                            </Typography>
                        </Grid>
                        <Grid item md={6}>
                            <FormControl component="fieldset">
                                <RadioGroup
                                    id="salary_rating"
                                    data-testid="salary_rating"
                                    name="salary_rating"
                                    row
                                    value={review.salary_rating}
                                    onChange={handleInput}
                                >
                                    <FormControlLabel value="Good" control={<Radio />} label="Good" />
                                    <FormControlLabel value="Regular" control={<Radio />} label="Regular" />
                                    <FormControlLabel value="Bad" control={<Radio />} label="Bad" />
                                </RadioGroup>
                            </FormControl>
                        </Grid>
                    </Grid>
                </Grid>
                <Grid item md={12} marginBottom={3}>
                    <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                        Job Location
                    </Typography>
                    <FormControl sx={{ width: '100%' }} required>
                        <OutlinedInput
                            id="job_location"
                            name="job_location"
                            placeholder="Job Location"
                            value={review.job_location}
                            onChange={handleInput}
                        />
                    </FormControl>
                </Grid>
                <Grid container marginBottom={2} spacing={2}>
                    <Grid item md={6}>
                        <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                            Start Date
                        </Typography>
                        <FormControl sx={{ width: '100%' }} required>
                            <DateInput
                                id="start_date"
                                name="start_date"
                                value={review.start_date}
                                onChange={handleInput}
                            />
                        </FormControl>
                    </Grid>
                    <Grid item md={6}>
                        <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                            End Date
                        </Typography>
                        <FormControl sx={{ width: '100%' }} required>
                            <DateInput
                                id="end_date"
                                name="end_date"
                                value={review.end_date}
                                disabled={!!review.is_still_working_here}
                                onChange={handleInput}
                            />
                        </FormControl>
                    </Grid>
                </Grid>
                <Grid container marginBottom={3}>
                    <FormControlLabel
                        control={<Checkbox onChange={handleCheck} checked={!!review.is_still_working_here} />}
                        label="Is still working here?"
                    />
                </Grid>
                <Grid container spacing={2}>
                    <Grid item md={6}>
                        <Grid item md={12} marginBottom={3}>
                            <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                                Whats the salary?
                            </Typography>
                            <FormControl sx={{ width: '40%' }} required>
                                <OutlinedInput
                                    id="salary"
                                    name="salary"
                                    placeholder="$"
                                    value={review.salary}
                                    onChange={handleInput}
                                />
                            </FormControl>
                            <FormControl sx={{ width: '30%' }} required>
                                <OutlinedInput
                                    id="currency_type"
                                    name="currency_type"
                                    placeholder="MXN"
                                    value={review.currency_type}
                                    onChange={handleInput}
                                />
                            </FormControl>
                            <FormControl sx={{ width: '30%' }} required>
                                <Select
                                    labelId="salary_frequency-label"
                                    id="salary_frequency"
                                    name="salary_frequency"
                                    value={review.salary_frequency}
                                    label="Per"
                                    onChange={handleInput}
                                >
                                    <MenuItem value="Year">Year</MenuItem>
                                    <MenuItem value="Month">Month</MenuItem>
                                    <MenuItem value="Day">Day</MenuItem>
                                    <MenuItem value="Hour">Hour</MenuItem>
                                </Select>
                            </FormControl>
                        </Grid>
                        <Grid item md={12}>
                            <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                                Would you recommend working here to a friend?
                            </Typography>
                            <FormControl component="fieldset" required>
                                <RadioGroup
                                    id="recommended_a_friend"
                                    name="recommended_a_friend"
                                    row
                                    value={review.recommended_a_friend}
                                    onChange={handleInput}
                                >
                                    <FormControlLabel value={1} control={<Radio />} label="Yes" />
                                    <FormControlLabel value={0} control={<Radio />} label="No" />
                                </RadioGroup>
                            </FormControl>
                        </Grid>
                    </Grid>
                    <Grid item md={6}>
                        <Grid item md={12} marginBottom={3}>
                            <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                                Does the job allows you to work from home?
                            </Typography>
                            <FormControl component="fieldset" required>
                                <RadioGroup
                                    id="remote_work_allowed"
                                    name="remote_work_allowed"
                                    row
                                    value={review.remote_work_allowed}
                                    onChange={handleInput}
                                >
                                    <FormControlLabel value={1} control={<Radio />} label="Yes" />
                                    <FormControlLabel value={0} control={<Radio />} label="No" />
                                </RadioGroup>
                            </FormControl>
                        </Grid>
                        <Grid item md={12}>
                            <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                                Is the company legally regulated?
                            </Typography>
                            <FormControl component="fieldset" required>
                                <RadioGroup
                                    id="is_legally_company"
                                    name="is_legally_company"
                                    row
                                    value={review.is_legally_company}
                                    onChange={handleInput}
                                >
                                    <FormControlLabel value={1} control={<Radio />} label="Yes" />
                                    <FormControlLabel value={0} control={<Radio />} label="No" />
                                </RadioGroup>
                            </FormControl>
                        </Grid>
                    </Grid>
                </Grid>
                {modalError && <Alert severity="error">{modalError}</Alert>}
                <Grid container sx={{ alignItems: 'center', justifyContent: 'center', marginTop: 5 }}>
                    <Grid item md={2} onClick={handleClose}>
                        <Button variant="outlined">Cancel</Button>
                    </Grid>
                    <Grid item md={2}>
                        <Button variant="contained" type="submit">
                            Submit
                        </Button>
                    </Grid>
                </Grid>
            </Box>
        </form>
    </Modal>
);

CompanyReviewForm.propTypes = {
    open: PropTypes.bool.isRequired,
    handleClose: PropTypes.func.isRequired,
    handleValidate: PropTypes.func.isRequired,
    modalError: PropTypes.string.isRequired,
    // eslint-disable-next-line react/forbid-prop-types
    review: PropTypes.object.isRequired,
    handleInput: PropTypes.func.isRequired,
    handleCheck: PropTypes.func.isRequired,
};

export default CompanyReviewForm;
