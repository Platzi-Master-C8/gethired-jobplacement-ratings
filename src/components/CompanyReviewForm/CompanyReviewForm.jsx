import React, { useState } from 'react';
import {
    FormControl,
    OutlinedInput,
    Grid,
    Typography,
    FormControlLabel,
    Box,
    Button,
    Modal,
    RadioGroup,
    Radio,
} from '@mui/material';
import CompanyReviewFormP2 from './CompanyReviewFormP2';

const boxStyles = {
    position: 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    width: '65%',
    height: '75%',
    bgcolor: 'background.paper',
    boxShadow: 24,
    p: 4,
    overflow: 'auto',
};

const CompanyReviewForm = () => {
    const [open, setOpen] = useState(false);

    const [review, setReview] = useState({
        companyId: 0,
        userId: 0,
        jobTitle: '',
        contentType: '',
        careerDevelopmentRating: '',
        diversityEqualOpportunityRating: '',
        workingEnvironmentRating: '',
        salaryRating: '',
        jobLocation: '',
        startDate: new Date().toISOString(),
        endDate: new Date().toISOString(),
        isStillWorkingHere: 0,
        salary: 0,
        salary_coin: '',
        salary_frequency: '',
        recommendedAFriend: 0,
        allowsRemoteWork: 0,
        isLegallyCompany: 0,
        utilityCounter: 0,
        nonUtilityCounter: 0,
        createdAt: new Date().toISOString(),
    });

    const handleOpen = () => setOpen(true);
    const handleClose = () => setOpen(false);

    const handleInput = (e) => {
        const { name, value } = e.target;
        setReview({ ...review, [name]: value });
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log(review);
        setOpen(false);
    };

    return (
        <div>
            <Grid md={12} sx={{ display: 'grid', justifyContent: 'flex-end' }}>
                <Button onClick={handleOpen}>Write a Review</Button>
            </Grid>
            <Modal
                open={open}
                onClose={handleClose}
                aria-labelledby="modal-modal-title"
                aria-describedby="modal-modal-description"
            >
                <form onSubmit={handleSubmit}>
                    <Box sx={boxStyles}>
                        <Typography variant="h1" align="center" sx={{ fontSize: 28 }}>
                            Review the Company
                        </Typography>
                        <Grid container marginTop={5}>
                            <Grid item md={12} marginBottom="20px">
                                <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                                    Job Title
                                </Typography>
                                <FormControl sx={{ width: '100%' }} required>
                                    <OutlinedInput
                                        id="jobTitle"
                                        name="jobTitle"
                                        placeholder="Job Title"
                                        value={review.jobTitle}
                                        onChange={handleInput}
                                    />
                                </FormControl>
                            </Grid>
                            <Grid item md={12} marginBottom="20px">
                                <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                                    Content
                                </Typography>
                                <FormControl sx={{ width: '100%' }} required>
                                    <OutlinedInput
                                        id="contentType"
                                        name="contentType"
                                        placeholder="Content"
                                        value={review.contentType}
                                        onChange={handleInput}
                                    />
                                </FormControl>
                            </Grid>
                        </Grid>
                        <Typography variant="h1" align="center" sx={{ fontSize: 28 }}>
                            Hoe would you rate the following areas ?
                        </Typography>
                        <Grid marginTop={3}>
                            <Grid container sx={{ alignItems: 'center' }}>
                                <Grid item md={6}>
                                    <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                                        Carrer development
                                    </Typography>
                                </Grid>
                                <Grid item md={6}>
                                    <FormControl component="fieldset" required>
                                        <RadioGroup
                                            id="careerDevelopmentRating"
                                            name="careerDevelopmentRating"
                                            row
                                            value={review.careerDevelopmentRating}
                                            onChange={handleInput}
                                        >
                                            <FormControlLabel value={0} control={<Radio />} label="Good" />
                                            <FormControlLabel value={1} control={<Radio />} label="Regular" />
                                            <FormControlLabel value={2} control={<Radio />} label="Bad" />
                                        </RadioGroup>
                                    </FormControl>
                                </Grid>
                            </Grid>
                            <Grid container sx={{ alignItems: 'center' }}>
                                <Grid item md={6}>
                                    <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                                        Diversity and Equal Opportunity
                                    </Typography>
                                </Grid>
                                <Grid item md={6}>
                                    <FormControl component="fieldset" required>
                                        <RadioGroup
                                            id="diversityEqualOpportunityRating"
                                            name="diversityEqualOpportunityRating"
                                            row
                                            value={review.diversityEqualOpportunityRating}
                                            onChange={handleInput}
                                        >
                                            <FormControlLabel value={0} control={<Radio />} label="Good" />
                                            <FormControlLabel value={1} control={<Radio />} label="Regular" />
                                            <FormControlLabel value={2} control={<Radio />} label="Bad" />
                                        </RadioGroup>
                                    </FormControl>
                                </Grid>
                            </Grid>
                            <Grid container sx={{ alignItems: 'center' }}>
                                <Grid item md={6}>
                                    <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                                        Working Environment
                                    </Typography>
                                </Grid>
                                <Grid item md={6}>
                                    <FormControl component="fieldset" required>
                                        <RadioGroup
                                            id="workingEnvironmentRating"
                                            name="workingEnvironmentRating"
                                            row
                                            value={review.workingEnvironmentRating}
                                            onChange={handleInput}
                                        >
                                            <FormControlLabel value={0} control={<Radio />} label="Good" />
                                            <FormControlLabel value={1} control={<Radio />} label="Regular" />
                                            <FormControlLabel value={2} control={<Radio />} label="Bad" />
                                        </RadioGroup>
                                    </FormControl>
                                </Grid>
                            </Grid>
                            <Grid container sx={{ alignItems: 'center' }}>
                                <Grid item md={6}>
                                    <Typography variant="subtitle1" sx={{ fontSize: 18 }}>
                                        Salary
                                    </Typography>
                                </Grid>
                                <Grid item md={6}>
                                    <FormControl component="fieldset" required>
                                        <RadioGroup
                                            id="salaryRating"
                                            name="salaryRating"
                                            row
                                            value={review.salaryRating}
                                            onChange={handleInput}
                                        >
                                            <FormControlLabel value={0} control={<Radio />} label="Good" />
                                            <FormControlLabel value={1} control={<Radio />} label="Regular" />
                                            <FormControlLabel value={2} control={<Radio />} label="Bad" />
                                        </RadioGroup>
                                    </FormControl>
                                </Grid>
                            </Grid>
                        </Grid>
                        <CompanyReviewFormP2 review={review} handleInput={handleInput} />
                        <Grid container sx={{ alignItems: 'center', justifyContent: 'center', marginTop: 5 }}>
                            <Grid item md={2} onClick={handleClose}>
                                <Button variant="outlined">Cancel</Button>
                            </Grid>
                            <Grid item md={2}>
                                <Button variant="outlined" type="submit">
                                    Submit
                                </Button>
                            </Grid>
                        </Grid>
                    </Box>
                </form>
            </Modal>
        </div>
    );
};

export default CompanyReviewForm;
