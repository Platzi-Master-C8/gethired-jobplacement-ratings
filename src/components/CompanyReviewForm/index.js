import CompanyReviewForm from './CompanyReviewForm';

export { CompanyReviewForm };
