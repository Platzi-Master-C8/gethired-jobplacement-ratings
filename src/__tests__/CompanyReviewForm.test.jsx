import React from 'react';

import { render, fireEvent } from '@testing-library/react';
import { CompanyReviewForm } from 'Components/CompanyReviewForm';

describe('Component <CompanyReviewForm/>', () => {
    test('renders CompanyReviewForm', () => {
        const { asFragment } = render(<CompanyReviewForm />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('it opens the modal', () => {
        const subject = render(<CompanyReviewForm />);

        expect(subject.container.querySelectorAll('h1').length).toBe(0);

        fireEvent.click(subject.getByText(/Write a Review/i));

        expect(subject.getByText(/Review the Company/i)).toBeInTheDocument();
    });
});
