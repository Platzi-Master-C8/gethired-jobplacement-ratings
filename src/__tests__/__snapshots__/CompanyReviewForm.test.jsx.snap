// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <CompanyReviewForm/> renders CompanyReviewForm 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-12 css-oav0mg-MuiGrid-root"
    >
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Write a Review
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
